src/palladio/Utils.cpp

```diff
@@ -356,4 +356,18 @@ void ensureNonExistingFile(std::filesystem::path& p) {
 	std::replace_if(
 	        stem.begin(), stem.end(), [](char c) { return (ILLEGAL_FS_CHARS.find(c) != std::string::npos); }, '_');
 	p = p.parent_path() / (stem + p.extension().generic_string());
+
+	// ensure we did not produce a collision with existing file
+	size_t suf = 0;
+	while (std::filesystem::exists(p)) {
+		if (suf == 0) {
+			stem += '_' + std::to_string(suf);
+		}
+		else if (auto sepPos = stem.rfind('_'); sepPos != std::string::npos) {
+			stem = stem.substr(0, sepPos) + '_' + std::to_string(suf);
```

Comment:
Maybe I just have a slight style preference, but wouldn't it make more sense to have the base '"stem" + "_"' as a separate std::string and then append the increasing suffix instead of rebuilding the string by making a char lookup and taking the substring to "revert" previous changes?
I think functionally it would be identical, to me it feels less error prone. But i am fine if you prefer to just use one variable..